File: Prototype/BigGp/src/Model/SmallNodes/Operators/Logic/BoolOperator.java
package Model.SmallNodes.Operators.Logic;

import Model.Node;
import Model.SmallNodes.Operators.Operator;
import Serialization.Token;

import java.util.ArrayList;

public class BoolOperator extends Operator {

    @Override
    public void mutate() {
        if (getValue().equals(" and ")) {
            setOperator(" or ");
        } else {
            setOperator(" and ");
        }
    }

    public BoolOperator(Node parentNode) {
        super(parentNode, "BoolOperator");

        if (getRandomInt(2) == 0) {
            setOperator(" and ");
        } else {
            setOperator(" or ");
        }
    }

    public BoolOperator(Node parentNode, ArrayList<Token> tokens) {
        super(parentNode, tokens);
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Small/Operators.Logic/BoolOperator.cs
﻿using Model.Abstract;
using Utils;

namespace Model.Nodes.Small.Operators.Logic;

public sealed class BoolOperator : Operator
{
    public override void Mutate()
    {
        Value = Value switch
        {
            " and " => " or ",
            " or " => " and ",
            _ => throw new InvalidOperationException()
        };
    }

    public BoolOperator(Node parentNode) : base(parentNode, "BoolOperator")
    {
        Value = RandomService.RandomPercentage() < 0.5
            ? " and "
            : " or ";
    }
    
    public BoolOperator(Node parentNode, IList<Token> tokens) : base(parentNode, tokens)
    {
    }
}